src/freenet/support/io/InetAddressIpv6FirstComparator.java

```diff
@@ -45,37 +45,37 @@ public int compare(InetAddress arg0, InetAddress arg1) {
 		} else if (arg0.isLinkLocalAddress() && !arg1.isLinkLocalAddress()) {
 			return 1;
 		}
-		// prefer reachable over unreachable addresses. This is usually a ping. TODO: This actually pings all advertised ip addresses. Is that OK? Do we need a maximum number of accepted addresses to prevent abuse as DDoS? 
+        // calculate hashCode only once; used multiple times later
 		int a = arg0.hashCode();
 		int b = arg1.hashCode();
-		Boolean reachable0 = reachabilityCache.get(a);
-		Boolean reachable1 = reachabilityCache.get(b);
-		if (reachable0 == null) {
-			try {
-				reachable0 = arg0.isReachable((int) DEFAULT_MAX_PING_TIME);
-			} catch (IOException e) {
-				reachable0 = false;
+		// prefer LAN addresses *if they are routable* over global addresses, because they should stay within VPNs
+		if (!arg0.isSiteLocalAddress() && arg1.isSiteLocalAddress()) {
```

Comment:
This does not not prefer reachable site-local addresses over other site-local addresses, as the comparison is only made when one of the addresses is global. Is that intentional?

Reply:
This is intentional, because this reachability check caused requests within VPNs to addresses outside the VPN and as such made a node much easier to detect. The reason is that Java falls back on non-standard requests when it cannot ping.